### PR TITLE
feat(datasource/azure-pipelines-tasks): add `azure-pipelines-tasks` datasource

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Adds the `azure-pipelines-tasks` datasource. 

Currently has only a static list of major versions of each built-in task[^1]. I outlined in the linked issue that not all versions of each task are available on every Azure DevOps organization, and an API call to your own Azure DevOps organization is required to find the specific versions available. Even when that is implemented, this static list of tasks will be required for repos that are hosted outside of Azure DevOps, but contain Azure Pipelines configurations.

TODO:
- [x] Run on a real repo
	- [`log.txt`](https://gist.github.com/JamieMagee/8f5212f6eb08c13638d77ffcaf4f3b0f)
	- PRs:
		- JamieMagee/renovate-azure-pipelines-tasks#1
		- JamieMagee/renovate-azure-pipelines-tasks#2
		- JamieMagee/renovate-azure-pipelines-tasks#3
		- JamieMagee/renovate-azure-pipelines-tasks#4
		-  JamieMagee/renovate-azure-pipelines-tasks#5
		-  JamieMagee/renovate-azure-pipelines-tasks#6
		-  JamieMagee/renovate-azure-pipelines-tasks#7
		-  JamieMagee/renovate-azure-pipelines-tasks#8
		-  JamieMagee/renovate-azure-pipelines-tasks#9
		-  JamieMagee/renovate-azure-pipelines-tasks#10
		-  JamieMagee/renovate-azure-pipelines-tasks#11
- [x] Update documentation

## Context

- Second step towards #15818
- Required by #16882

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

[^1]: https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks

### CODE REVIEW

**Comment by viceice** [comment]
You should add the limitations to the datasource readme

**Comment by JamieMagee** [comment]
> You should add the limitations to the datasource readme

It was on my TODO list 😉 

Now it's on my TODONE list

File: lib/modules/datasource/azure-pipelines-tasks/built-in.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+export const BuiltInAzurePipelinesTaskMajorVersions: Map<string, string[]> =
```
Comment (rarkins) [inline lib/modules/datasource/azure-pipelines-tasks/built-in.ts:1]: Where does this list come from and how should it be maintained in future?

Comment (JamieMagee) [reply lib/modules/datasource/azure-pipelines-tasks/built-in.ts:1]: They come from [this directory](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks). I'm trying to identify an API that will provide the same data.

Comment (JamieMagee) [reply lib/modules/datasource/azure-pipelines-tasks/built-in.ts:1]: @rarkins I wrote a script to keep this up-to-date #16949 16949

**Comment by JamieMagee** [comment]
I rebased this branch on top of #16949, re-ran Renovate, and generated some new PRs.

File: lib/modules/datasource/azure-pipelines-tasks/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { getPkgReleases } from '../index';
```
Comment (viceice) [inline lib/modules/datasource/azure-pipelines-tasks/index.spec.ts:1]:
```suggestion
import { getPkgReleases } from '..';
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**